.github/workflows/Windows-CI.yml

```
@@ -107,7 +107,7 @@ jobs:
 
         # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
       - name: checkout
-        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 #v3.5.3
+        uses: actions/checkout@v4.1.1
```

Can we consider using the full SHA, together with a comment identifying which version it is? I.e.:

```suggestion
        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
```

---

SHA is harder to lookup. Is there a specific reason? Using the tag allows easier updates.

---

Ahh... https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash got it

---

@BenjamenMeyer Yep.

---

The key seems to be to find the source repository for the GitHub Action in question. Usually there will be a link to it in the sidebar of the GH Action's Marketplace page. (E.g., https://github.com/marketplace/actions/ossf-scorecard-action links to https://github.com/ossf/scorecard-action ) Then you can look at the Releases (or sometimes just Tags) for that repo; find the latest one; and copy/paste its hash.

If that makes sense.

---

updated the rest of them

---

Nice! Looks good!